schema/generate_scaler_schema.go

```diff
@@ -0,0 +1,552 @@
+		if err != nil {
+			return err
+		}
+
+		filedata = append(filedata, '\n')
+		fileName = outputFilePath + "/" + outputFileName + ".json"
+		err = os.WriteFile(fileName, filedata, 0644)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("output file format %s is not supported", outputFileFormat)
+	}
+
+	return err
+}
+
+// generateMetadataFields is a function that generates the metadata fields of a scaler struct
+func generateMetadataFields(structType *ast.StructType, otherReferenceKedaTagStructs map[string]*ast.StructType) []Metadata {
+	scalerMetadata := []Metadata{}
+
+	// get the tag of each field and generate the metadata
+	for _, commentGroup := range structType.Fields.List {
+		if commentGroup.Tag == nil || commentGroup.Tag.Value == "" {
+			continue
+		}
+		metadataList, hasSubstruct, err := generateMetadatas(commentGroup.Tag.Value)
+
+		if err != nil {
+			fmt.Printf("Error generating metadata fields from tag value: %s, err: %s\n", commentGroup.Tag.Value, err)
+			continue
+		}
+
+		if !hasSubstruct {
+			scalerMetadata = append(scalerMetadata, metadataList...)
+			continue
+		}
+
+		// If the field has a substruct, try to find substruct from reference structs
+		s, ok := commentGroup.Type.(*ast.Ident)
+		if !ok {
+			continue
+		}
+		if otherReferenceKedaTagStructs[s.Name] != nil {
+			subStructMetadataField := generateMetadataFields(otherReferenceKedaTagStructs[s.Name], otherReferenceKedaTagStructs)
+			if len(subStructMetadataField) > 0 {
+				scalerMetadata = append(scalerMetadata, subStructMetadataField...)
+			}
+		}
+	}
+
+	return scalerMetadata
+}
+
+// generateMetadatas is a function that generates the metadata field from tag
+func generateMetadatas(tag string) ([]Metadata, bool, error) {
+	var fieldNames []string
+	metadata := Metadata{Type: "string"}
+	tagSplit := strings.Split(strings.Trim(strings.Split(strings.Trim(tag, "`"), ":")[1], "\""), scalersconfig.TagSeparator)
+
+	if len(tagSplit) == 1 && tagSplit[0] == scalersconfig.OptionalTag {
+		return nil, true, nil
+	}
+
+	for _, ts := range tagSplit {
+		tsplit := strings.Split(ts, scalersconfig.TagKeySeparator)
+		tsplit[0] = strings.TrimSpace(tsplit[0])
+		switch tsplit[0] {
+		case scalersconfig.OptionalTag:
+			if len(tsplit) == 1 {
+				metadata.Optional = true
+			}
+			if len(tsplit) > 1 {
+				optional, err := strconv.ParseBool(strings.TrimSpace(tsplit[1]))
+				if err != nil {
+					return nil, false, fmt.Errorf("error parsing optional value %s: %w", tsplit[1], err)
+				}
+				metadata.Optional = optional
+			}
+		case scalersconfig.OrderTag:
+			if len(tsplit) > 1 {
+				order := strings.Split(tsplit[1], scalersconfig.TagValueSeparator)
+				canReadFromEnv, canReadFromAuth, err := retrieveDataFromOrder(order)
+				if err != nil {
+					return nil, false, err
+				}
+				metadata.EnvVariableReadable = canReadFromEnv
+				metadata.TriggerAuthenticationVariableReadable = canReadFromAuth
+			}
```

Comment:
Checking how the generated files look, I think that we should use different sections for each source or maybe rename the current `metadata` by `parameters` and add another field `MetadataVariableReadable`. 
Currently there are some parameters (like tokens) which only can be read from secure sources and with this approach, they look as readable from metadata when that's not correct

Reply:
For example, after fixing dynatrace scaler it looks like:
```yaml
- type: dynatrace
      metadata:
        - name: token
          type: string
          triggerAuthenticationVariableReadable: true
...
```
This seems that token can be set as metadata when it can only be set as auth, my suggestion is to expose the things like:
```yaml
- type: dynatrace
      parameters:
        - name: host
          type: string
          triggerAuthenticationVariableReadable: true
          metadataVariableReadable: true
        - name: token
          type: string
          triggerAuthenticationVariableReadable: true
        - name: metricSelector
          type: string
          metadataVariableReadable: true
...
```